modules/utility/binds/which-key/which-key.nix

```diff
@@ -1,7 +1,77 @@
-{lib, ...}: let
-  inherit (lib) mkEnableOption;
+{
+  config,
+  lib,
+  ...
+}: let
+  inherit (lib) mkEnableOption mkOption mkOptionType types;
+
+  whichKeyCategory = mkOptionType {
+    name = "whichKeyCategory";
+    description = "The label to use for the which-key keybind helper menu.";
+    check = label:
+      if label == null && !(lib.isAttr label)
+      then false
+      else if lib.hasAttr "name" label && lib.isString label.name
+      then true
+      else false;
+  };
 in {
   options.vim.binds.whichKey = {
     enable = mkEnableOption "which-key keybind helper menu";
+
+    register = mkOption {
+      description = "Register label for which-key keybind helper menu";
+      type = types.attrsOf whichKeyCategory;
+      default =
```

Comment:
a mkMerge would be better here

```nix

mkMerge [
	(mkIf config.vim.tabline.nvimBufferline.enable {
		"<leader>b" = {name = "+Buffer";};
  	"<leader>bm" = {name = "BufferLineMove";};
    "<leader>bs" = {name = "BufferLineSort";};
    "<leader>bsi" = {name = "BufferLineSortById";};
   }) 
]

Reply:
The reason I preferred using attrset and `//` operator is because the check function in mkOptionType expects:
```nix
{ name = "+Telescope"; }
```
instead of:
```nix
{ "<leader>f" = { name = "+Telescope"; }; }
```
Which simplifies the validation because it is not trying to extract arbitrary named elements from the anonymous attrset. Resolving this for now and leaving this for the future. 🤞

Reply:
Surely using an existing option type would make more sense than rolling your own here? Then you don't need to write your own `check` function.

I would normally use `attrsOf submodule` here.

Reply:
Indeed, I was just unaware about the existence of submodules. Thank you 👍